### PR TITLE
Legalize parametric patch names

### LINKED ISSUE #1114: Legalize parametric patch names
### Rationale

To support generics we should allow patch names which define the type(s) they specialize for:

- `if-else(number)`
- `if-else(string)`
- `cast(boolean,pulse)`

### Acceptance criteria

The following are legal and can be created via IDE:

- [ ] `if-else(number)`
- [ ] `cast(boolean,pulse)`

The following is not:

- [ ] `if-else(number`
- [ ] `if-else ( number )`
- [ ] `if-else)`
- [ ] `if-else()`
- [ ] `if-else(number)string`

### Out of scope

- Any special meaning for the patches with parenthesis in their name.
- Fancy rendering of parametric patch names

### PR DESCRIPTION
**Pull request by brusherru**
It closes #1114.

Also tweaked rename popup. Now input already contains current patch name and autoselected.

### CODE REVIEW

File: packages/xod-project/src/internal/patchPathUtils.js [hunk 1]
```diff
@@ -8,13 +8,31 @@ export const getExpandedVariadicPatchPath = R.curry(
     [originalPath, EXPANDED_PATCH_NAME_SUFFIX, arityLevel].join('')
 );
 
+// =============================================================================
+// RegExp parts
+// =============================================================================
+const alphanumeric = '[a-z0-9]+';
+const alphanumericWithHypens = `${alphanumeric}(-${alphanumeric})*`;
+const alphanumericWithHypensAndCommans = `${alphanumeric}(-${alphanumeric}|,${alphanumeric})*`;
+const variadicLevel = '-\\$\\d';
```
Comment (nkrkv) [inline packages/xod-project/src/internal/patchPathUtils.js]: I love the idea, but let’s remove this easter egg until we’ll make the resolution for the different variadic levels?

File: packages/xod-project/src/internal/patchPathUtils.js [hunk 2]
```diff
@@ -8,13 +8,31 @@ export const getExpandedVariadicPatchPath = R.curry(
...
+const types = `(\\(${alphanumericWithHypensAndCommans}\\)){0,1}`;
+
+const patchBaseNameRegExp = new RegExp(
+  `^${alphanumericWithHypens}(${types}|${variadicLevel})$`
+);
+const identifierRegExp = new RegExp(`^${alphanumericWithHypens}$`);
```
Comment (nkrkv) [inline packages/xod-project/src/internal/patchPathUtils.js]: Cool readable regex constructor :+1:

File: packages/xod-project/test/patchPathUtils.spec.js [hunk 3]
```diff
@@ -8,6 +8,11 @@ import * as Helper from './helpers';
 
 chai.use(dirtyChai);
 
+// Kludge to use instead of "assert.isTrue" / "assert.isFalse"
+// Cause dirtyChai makes it works wrong
+const isTrue = a => assert.equal(a, true);
+const isFalse = a => assert.equal(a, false);
```
Comment (nkrkv) [inline packages/xod-project/test/patchPathUtils.spec.js]: I filed #1129. To drop the kludge easily once the refactoring will land, maybe it is better to over-monkey-patch `assert.isTrue` and `assert.isFalse` right here and remove the `dirtyChai` from this particular module?

If agree, lets make it and leave a TODO referring the #1129

File: packages/xod-project/src/internal/patchPathUtils.js (hunk 1 above)
Comment (brusherru) [reply packages/xod-project/src/internal/patchPathUtils.js]: It's just about PatchPath type, to validate flattened variadic nodes.
Early validation of patch paths did not work and so this rule appeared only now.

**Pull request merged**